src/views/Dashboard.vue

```diff
@@ -0,0 +1,249 @@
+	- published by the Free Software Foundation, either version 3 of the
+	- License, or (at your option) any later version.
+	-
+	- This program is distributed in the hope that it will be useful,
+	- but WITHOUT ANY WARRANTY; without even the implied warranty of
+	- MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+	- GNU Affero General Public License for more details.
+	-
+	- You should have received a copy of the GNU Affero General Public License
+	- along with this program. If not, see <http://www.gnu.org/licenses/>.
+	-
+	-->
+
+<template>
+	<div v-if="events === null">
+		<div
+			v-for="index in 5"
+			:key="index"
+			class="placeholder-line">
+			<div class="placeholder-icon" />
+			<div class="placeholder-text" />
+		</div>
+	</div>
+	<EmptyContent v-else-if="events.length === 0">
+		<img alt="" :src="imagePath">
+		{{ t('calendar', 'Nothing upcoming for now') }}
+		<template #desc>
+			<p />
+			<button>{{ t('calendar', 'Create a new event') }}</button>
+		</template>
+	</EmptyContent>
+	<div v-else>
+		<ul>
+			<li v-for="event in events" :key="event.id">
+				<a :href="event.deepLink" class="event">
+					<div class="event__title">
+						<div
+							class="event__calendar-dot"
+							:title="event.calendarTitle"
+							:style="{'background-color': event.calendarColor}" />
+						<strong>{{ event.title }}</strong>
+					</div>
+					<div class="event__details">{{ event.startDate }}</div>
+				</a>
+			</li>
+		</ul>
+	</div>
+</template>
+
+<script>
+import { imagePath, generateUrl } from '@nextcloud/router'
+import EmptyContent from '@nextcloud/vue/dist/Components/EmptyContent'
+import { initializeClientForUserView } from '../services/caldavService'
+import { dateFactory } from '../utils/date'
+import pLimit from 'p-limit'
+import { eventSourceFunction } from '../fullcalendar/eventSourceFunction'
+import getTimezoneManager from '../services/timezoneDataProviderService'
+import loadMomentLocalization from '../utils/moment.js'
+import moment from '@nextcloud/moment'
+import { loadState } from '@nextcloud/initial-state'
+
+export default {
+	name: 'Dashboard',
+	components: {
+		EmptyContent,
+	},
+	data() {
+		return {
+			events: null,
+			imagePath: imagePath('calendar', 'illustrations/calendar'),
+		}
+	},
+	mounted() {
+		setTimeout(() => this.initialize(), 1000)
+	},
+	methods: {
+		/**
+		 * Initialize the widget
+		 *
+		 * @returns {Promise<void>}
+		 */
+		async initialize() {
+			await initializeClientForUserView()
+			await this.$store.dispatch('fetchCurrentUserPrincipal')
+			const locale = await loadMomentLocalization()
+			const calendars = await this.$store.dispatch('getCalendars')
+
+			const {
+				show_tasks: showTasks,
+				timezone,
+			} = loadState('calendar', 'dashboard_data')
+			this.$store.commit('loadSettingsFromServer', {
+				timezone,
+				showTasks,
+			})
+
+			const start = dateFactory()
+			const end = dateFactory()
+			end.setDate(end.getDate() + 14)
```

Comment:
@jancborchardt Should it be 14 days or end of next week?

Reply:
14 days is good, that way there’s always 2 weeks of relevant events. :)